lib/validation.rb

```diff
@@ -173,6 +173,23 @@ def self.validate_port_range(port_range)
     end_port ? [start_port, end_port] : [start_port]
   end
 
+  def self.validate_port(port_name, port)
+    fail ValidationFailed.new({port_name => "Port must be between 0 to 65535"}) unless port.to_i.to_s == port
+    fail ValidationFailed.new({port_name => "Port must be between 0 to 65535"}) unless (0..65535).cover?(port.to_i)
+    port.to_i
+  end
+
+  def self.validate_health_check_parameter(parameter_name, parameter_value)
+    unless validate_positive_int(parameter_value)
+      fail ValidationFailed.new({parameter_name => "Value must be a positive integer."})
+    end
+    parameter_value.to_i
+  end
+
+  def self.validate_positive_int(parameter)
+    parameter.to_i.to_s == parameter && parameter.to_i > 0
+  end
```

Comment:
Our validation methods raise exception if the validation failed. `validate_positive_int` looks like antipattern. Since it's used in a single place, we can it in that place. Or you can rename it as something like `self.positive_int?` 

```suggestion
  def self.validate_health_check_parameter(name, value)
    unless value.to_i.to_s == value && value.to_i > 0
      fail ValidationFailed.new({name => "Value must be a positive integer."})
    end
    value.to_i
  end
```